node/services/bn-manager.go

```diff
@@ -73,10 +79,49 @@ func (m *BeaconClientManager) GetClientTypeName() string {
 
 func (m *BeaconClientManager) SetPrimaryReady(ready bool) {
 	m.primaryReady = ready
+	if ready {
+		m.primaryFailTime = time.Time{}
+	} else {
+		m.primaryFailTime = time.Now()
+	}
 }
 
 func (m *BeaconClientManager) SetFallbackReady(ready bool) {
 	m.fallbackReady = ready
+	if ready {
+		m.fallbackFailTime = time.Time{}
+	} else {
+		m.fallbackFailTime = time.Now()
+	}
+}
+
+func (m *BeaconClientManager) RecheckFailTimes(logger *log.Logger) {
```

Comment:
Might be better to name this function something that contains the word 'throttle', since it throttles reconnection attempts to `recheckTime` intervals. Might even be sensible to return a boolean to indicate whether or not a reconnection was attempted.